test/remote/README.md

```diff
@@ -0,0 +1,240 @@
+
+### Build the repository container image
+
+You will need to build & publish a container image that will hold the required
+testing repositories.
+
+This container image will be named `argocd-e2e-cluster`, so you will need to
+setup a corresponding repository for it in your registry as well.
+
+To build it, run the following. Note that kustomize is required:
+
+```shell
+cd test/remote
+export IMAGE_NAMESPACE=quay.io/youruser
+# builds & tags the image
+make image
+# pushes the image to your repository
+make image-push
+# build the manifests & store them at temp location
+make manifests > /tmp/e2e-repositories.yaml
+```
+
+If you do not have kustomize installed, you need to manually edit the manifests
+at `test/remote/manifests/e2e_repositories.yaml` to point to the correct image.
+
+### Deploy the test container and additional permissions
+
+**Note:** The test container requires to be run in privileged mode for now, due
+to some processes running as root (this may change some day...).
+
+On OpenShift, you will likely need to allow privileged operations:
+
+```shell
+oc -n argocd-e2e adm policy add-scc-to-user privileged -z default
+```
+
+Then, apply the manifests for the E2E repositories workload:
+
+```shell
+kubectl -n argocd-e2e apply -f /tmp/e2e-repositories.yaml
+```
+
+Verify that the deployment was succesful:
+
+```shell
+kubectl -n argocd-e2e rollout status deployment argocd-e2e-cluster
+```
+
+## Start the tests
+
+### Port-forward the repository
+
+In another shell, port forward the repository to your local machine:
+
+```shell
+kubectl -n argocd-e2e port-forward service/argocd-e2e-server 9081:9081
+```
+
+### On local cluster (e.g. K3s, microk8s, minishift)
+
+Set the server endpoint of the Argo CD API. If you are running on the same host
+as the cluster, or the cluster IPs are routed to your host, you can use the
+following:
+
+```shell
+export ARGOCD_SERVER=$(kubectl get svc argocd-server -o jsonpath='{.spec.clusterIP}')
+```
+
+Set the admin password to use:
+
+```shell
+export ARGOCD_E2E_ADMIN_PASSWORD=$(kubectl get secrets argocd-initial-admin-secret -o jsonpath='{.data.password}'|base64 -d)
+```
+
+Run the tests
+
+```shell
+./test/remote/run-e2e-remote.sh make test-local
+```
+
+### On remote non-OpenShift cluster
+
+In another shell, do a port-forward to the API server's service:
+
+```shell
+kubectl -n argocd-e2e port-forward svc/argocd-server 443:4443
+```
+
+Set Argo CD Server endport:
+
+```shell
+export ARGOCD_SERVER=127.0.0.1:4443
+```
+
+Set the admin password to use:
+
+```shell
+export ARGOCD_E2E_ADMIN_PASSWORD=$(kubectl get secrets argocd-initial-admin-secret -o jsonpath='{.data.password}'|base64 -d)
+```
+
+Run the tests
+
+```shell
+./test/remote/run-e2e-remote.sh make test-local
+```
+
+### On remote OpenShift cluster
+
+You should first scale down GitOps Operator, since it will revert changes made
+during the tests instantly:
+
+```shell
+oc -n openshift-operators scale deployment --replicas=0 gitops-operator
```

Comment:
Suggested replacement for upstream:

/s/gitops-operator/argocd-operator :)

Reply:
Done. Thanks!